network/src/protocols/discovery/mod.rs

```diff
@@ -132,6 +134,9 @@ impl<M: AddressManager> ServiceProtocol for DiscoveryProtocol<M> {
                                 if let RemoteAddress::Listen(ref addr) = state.remote_addr {
                                     self.addr_mgr.add_new_addr(session.id, addr.clone());
                                 }
+                            } else if version == state::REUSE_PORT_VERSION {
```

Comment:
we should consider future version upgrade
```suggestion
                            } else if version >= state::REUSE_PORT_VERSION {
```